packages/node_modules/@webex/plugin-meetings/src/locus-Info/parser.js

```diff
@@ -0,0 +1,114 @@
+import LocusUtil from '../locus-events/utils';
+import {DELTAEVENT} from '../constants';
+
+const Parser = {};
+
+// https://sqbu-github.cisco.com/WebExSquared/cloud-apps/wiki/Locus-Delta-Events
+Parser.updateWithDeltaLocus = (deltaLocus, meeting) => {
+  // /* istanbul ignore else */
+  // if (!(deltaLocus.sequence && (deltaLocus.sequence.rangeStart <= deltaLocus.sequence.rangeEnd) && (deltaLocus.sequence.rangeEnd <= min(deltaLocus.sequence.entries)))) {
+  //   return null;
+  // }
+  // /* istanbul ignore else */
+
+  // THIS IS A NEW MEETING OBJECT
+  // // If meeting has no sequence, it means just initialized and will apply whatever in locus DTO
+  // if (isNewMeeting || !this.locus.sequence) {
+  //   meeting._updateWithDeltaLocus(deltaLocus, emitter);
+  //   meeting.locusDesync = false;
+  //   meeting.needToGetFullLocus = false;
+  //   return DELTAEVENT.GT;
+  // }
+
+  if (!deltaLocus.sequence) {
+    // TODO: FORce update
+  }
+
+  // TODO: If this was the first event, the client will not have a syncURL
+  // in its working copy to do a GET on. In this case clients should do a GET on locusUrl contained in the DTO
+
+  let res;
+  let resBase;
+  res = LocusUtil.compareLocus(deltaLocus.sequence, meeting.locus.sequence);
+  // In-coming locus is greater than working copy, need to compare with base if there
+  // is base sequence except the first delta locus event
+  if (res === DELTAEVENT.GT && deltaLocus.baseSequence && meeting.locus.syncUrl) {
+    resBase = LocusUtil.compareLocus(deltaLocus.baseSequence, meeting.locus.sequence);
+    if (resBase === DELTAEVENT.LT || resBase === DELTAEVENT.CF) {
+      res = DELTAEVENT.CF;
+    }
+  }
+  // Special case for OBTP with meeting size equal or over 5. Locus might change this later
+  // This is a temporary fix. I will find a better solution later.
```

Comment:
can we get some more notes on this?

Reply:
benrui added it it. we will change it when we implement obtp